types/hapi/definitions/route/route-options-response.d.ts

```diff
@@ -0,0 +1,84 @@
+import {Lifecycle, Util} from "hapi";
+import {ValidationOptions} from "joi";
+
+/**
+ * For context [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-routeoptionsresponseemptystatuscode)
+ */
+export enum EmptyStatusCode {
+    HTTP_200 = 200,
+    HTTP_204 = 204,
+}
+
+export type RouteOptionsResponseSchema = boolean | ValidationOptions | ((value: object, options: object) => any);
```

Comment:
Should this also include an option for a joi object?  Something like 
```typescript
export type RouteOptionsResponseSchema = boolean | ValidationOptions | 
  ((value: object, options: object) => any) | {[s: string]: JoiObject} | ObjectSchema;
```

Without this, you will get a type error on [this example](https://hapijs.com/tutorials/validation?lang=en_US) from the hapi docs:
```javascript
server.route({
    method: 'GET',
    path: '/hello/{name}',
    handler: function (request, reply) {
        reply('Hello ' + request.params.name + '!');
    },
    config: {
        validate: {
            params: {
                name: Joi.string().min(3).max(10)
            }
        }
    }
});
```

Reply:
@ajmath I added a test file for this case 19a1b22. I think in this case ```server.route(object)``` the object.config is the ServerRouteConfig, that is empty.

```
export interface ServerRouteConfig {
}
```
Can you review the test case please? Thanks!